module/optee/mbx/test/mod_optee_mbx_unit_test.c

```diff
@@ -27,10 +30,369 @@ void tearDown(void)
 {
 }
 
+struct mbx_device_ctx fake_optee_mbx_ctx[FAKE_SCMI_SERVICE_IDX_COUNT];
+
+/*!
+ * \brief test module's init function.
+ *
+ *  \details Test all possible cases for init.
+ */
+void test_optee_mbx_init(void)
+{
+    int status;
+    fwk_id_t init_id;
+
+    init_id = FWK_ID_MODULE(FWK_MODULE_IDX_OPTEE_MBX);
+
+    /* Element count can't be null */
+    status = mbx_init(init_id, 0, NULL);
+    TEST_ASSERT_EQUAL(FWK_E_PARAM, status);
+
+    /* Fail to allocate module cxt */
+    fwk_mm_calloc_ExpectAndReturn(
+        1,
+        sizeof(struct mbx_device_ctx),
+        NULL);
+    status = mbx_init(init_id, FAKE_SCMI_SERVICE_IDX_COUNT, NULL);
+    TEST_ASSERT_EQUAL(FWK_E_NOMEM, status);
+
+    /* Allocate module context */
+    fwk_mm_calloc_ExpectAndReturn(
+        FAKE_SCMI_SERVICE_IDX_COUNT,
+        sizeof(struct mbx_device_ctx),
+        fake_optee_mbx_ctx);
+    status = mbx_init(init_id, FAKE_SCMI_SERVICE_IDX_COUNT, NULL);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+    TEST_ASSERT_EQUAL(mbx_ctx.device_ctx_table, fake_optee_mbx_ctx);
+}
+
+/*!
+ * \brief test module's element_init function.
+ *
+ *  \details Test all possible cases for element_init.
+ */
+void test_optee_mbx_device_init(void)
+{
+    int status;
+    fwk_id_t device_id;
+    struct mod_optee_mbx_channel_config *device_config;
+
+    /* Set fake config data */
+    device_config =
+        (struct mod_optee_mbx_channel_config *)mbx_element_table[FAKE_SCMI_SERVICE_IDX_OSPM_0]
+            .data;
+
+    /* Successfully init element */
+    device_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_MBX, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(device_id, 0);
+    mutex_init_ExpectAnyArgs();
+    status = mbx_device_init(device_id, 0, device_config);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+}
+
+/*!
+ * \brief test module's bind fucntion for round 0.
+ *
+ *  \details Test bind for round 0 and others except for round 1 when it needs
+ *  additional information from bind_request of other modules.
+ */
+void test_optee_mbx_bind_round0(void)
+{
+    int status;
+    fwk_id_t bind_id;
+    unsigned int i;
+
+    /*
+     * Bind module always succeed because there is nothing to do ...
+     * Take the opportunity to test all possible rounds in one shot because
+     * it's a nop.
+     */
+    bind_id = FWK_ID_MODULE(FWK_MODULE_IDX_OPTEE_MBX);
+    for (i = 0; i <= FWK_MODULE_BIND_ROUND_MAX; i++) {
+        fwk_id_is_type_ExpectAndReturn(bind_id, FWK_ID_TYPE_MODULE, true);
```

Comment:
I see in your 3rd https://github.com/ARM-software/SCP-firmware/pull/834
that you tested for the -ve case here 
is it possible to do something similar here?